packages/core/src/routes/interaction/middleware/koa-interaction-hooks.ts

```diff
@@ -1,31 +1,79 @@
-import { trySafe } from '@silverhand/essentials';
+import { type InteractionEvent } from '@logto/schemas';
+import { conditional, trySafe } from '@silverhand/essentials';
 import type { MiddlewareType } from 'koa';
 import type { IRouterParamContext } from 'koa-router';
 
-import type TenantContext from '#src/tenants/TenantContext.js';
+import type Libraries from '#src/tenants/Libraries.js';
 
 import { getInteractionStorage } from '../utils/interaction.js';
 
 import type { WithInteractionDetailsContext } from './koa-interaction-details.js';
 
+export type InteractionHookContext = {
+  event: InteractionEvent;
+  sessionId?: string;
+  applicationId?: string;
+};
+
+export type InteractionHookResults = {
+  userId: string;
+};
+
+type AssignInteractionHookDetails = (details: InteractionHookResults) => void;
+
+export type WithInteractionHooksContext<
+  ContextT extends IRouterParamContext = IRouterParamContext
+> = ContextT & { assignInteractionHookResults: AssignInteractionHookDetails };
+
+/**
+ * The factory to create a new interaction hook middleware function for the submit interaction route.
+ * Submit interaction related event hooks will be triggered once we got the interaction hook results.
+ * Use `assignInteractionHookResults` to assign the interaction hook results.
```

Comment:
I don't think it's designed for the submit interaction route only. on the contrary, the purpose of this middleware function is to decouple the triggering logic from a specific route. rephrase this part.